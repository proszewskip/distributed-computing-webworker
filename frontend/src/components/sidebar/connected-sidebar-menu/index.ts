export * from './connected-sidebar-menu';
export { ConnectedMenuItem, ConnectedSidebarMenuProps } from './types';
