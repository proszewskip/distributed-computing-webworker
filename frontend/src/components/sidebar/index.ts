export * from './sidebar-menu';
export * from './sidebar';
export * from './connected-sidebar-menu';
