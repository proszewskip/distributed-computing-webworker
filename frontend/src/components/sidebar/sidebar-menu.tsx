import { Button, Pane } from 'evergreen-ui';
import Link, { LinkProps } from 'next/link';
import React, { StatelessComponent } from 'react';

export interface MenuItem {
  title: string;
  href: LinkProps['href'];
  isActive: () => boolean;
}

export interface SidebarMenuProps {
  items: MenuItem[];
}

export const SidebarMenu: StatelessComponent<SidebarMenuProps> = (props) => {
  return (
    <Pane display="flex" flexDirection="column" alignItems="stretch">
      {props.items.map((item) => (
        <Link href={item.href} passHref={true} key={item.title}>
          <Button is="a" appearance="minimal" isActive={item.isActive()}>
            {item.title}
          </Button>
        </Link>
      ))}
    </Pane>
  );
};
