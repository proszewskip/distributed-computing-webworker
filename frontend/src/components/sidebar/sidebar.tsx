import { Heading, minorScale, Pane } from 'evergreen-ui';
import React, { ReactNode, StatelessComponent } from 'react';

export interface SidebarProps {
  title: ReactNode;
}

/**
 * The layout for the sidebar.
 */
export const Sidebar: StatelessComponent<SidebarProps> = ({
  title,
  children,
}) => {
  return (
    <Pane
      display="flex"
      flexDirection="column"
      background="tint2"
      height="100%"
    >
      <Heading
        size={600}
        marginTop="default"
        marginBottom={minorScale(2)}
        paddingX={minorScale(3)}
      >
        {title}
      </Heading>

      {children}
    </Pane>
  );
};
