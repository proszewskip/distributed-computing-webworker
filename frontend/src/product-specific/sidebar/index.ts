export * from './authenticated-sidebar';
export * from './authenticated-sidebar-menu-items';
