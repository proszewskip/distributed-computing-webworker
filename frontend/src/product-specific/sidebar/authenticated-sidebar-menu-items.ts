import { ConnectedMenuItem } from 'components/sidebar';

export const authenticatedSidebarMenuItems: ConnectedMenuItem[] = [
  {
    route: '/',
    title: 'Home',
  },
  {
    route: '/distributed-task-definitions',
    title: 'Distributed Task Definitions',
  },
  {
    route: '/distributed-nodes',
    title: 'Distributed Nodes',
  },
];
