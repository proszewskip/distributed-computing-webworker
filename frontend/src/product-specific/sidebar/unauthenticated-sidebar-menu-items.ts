import { ConnectedMenuItem } from 'components/sidebar';

export const unauthenticatedSidebarMenuItems: ConnectedMenuItem[] = [
  {
    route: '/',
    title: 'Home',
  },
  {
    route: '/worker',
    title: 'Worker',
  },
  {
    route: '/authentication/login',
    title: 'Login',
  },
];
