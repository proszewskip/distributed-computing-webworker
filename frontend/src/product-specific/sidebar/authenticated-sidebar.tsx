import React, { StatelessComponent } from 'react';

import { ConnectedSidebarMenu, Sidebar } from 'components/sidebar';
import { authenticatedSidebarMenuItems } from './authenticated-sidebar-menu-items';

export const AuthenticatedSidebar: StatelessComponent = () => (
  <Sidebar title="Distributed Computing">
    <ConnectedSidebarMenu items={authenticatedSidebarMenuItems} />
  </Sidebar>
);
