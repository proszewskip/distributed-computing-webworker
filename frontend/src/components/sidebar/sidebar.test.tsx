import React from 'react';
import renderer from 'react-test-renderer';

import { Sidebar } from './sidebar';

describe('Sidebar', () => {
  it('should match snapshot', () => {
    const instance = renderer.create(
      <Sidebar title="Bar">
        <div>Foo</div>
      </Sidebar>,
    );

    expect(instance.toJSON()).toMatchSnapshot();
  });
});
