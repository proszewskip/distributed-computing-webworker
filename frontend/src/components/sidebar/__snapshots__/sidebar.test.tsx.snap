// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match snapshot 1`] = `
<div
  className="ðŸ“¦bg_1eozcqw ðŸ“¦dspl_flex ðŸ“¦flx-drct_column ðŸ“¦h_100prcnt ðŸ“¦box-szg_border-box css-nil"
>
  <h2
    className="ðŸ“¦mt_28px ðŸ“¦mb_8px ðŸ“¦fnt-sze_20px ðŸ“¦f-wght_500 ðŸ“¦ln-ht_24px ðŸ“¦ltr-spc_-0-07px ðŸ“¦fnt-fam_wm9v6k ðŸ“¦color_234361 ðŸ“¦pl_12px ðŸ“¦pr_12px ðŸ“¦box-szg_border-box"
  >
    Bar
  </h2>
  <div>
    Foo
  </div>
</div>
`;
